core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -158,6 +163,26 @@ public KafkaSchemaRegistry(SchemaRegistryConfig config,
     this.masterNodeSensor.add(m, new Gauge());
   }
 
+  /**
+   * A Schema Registry instance's identity is in part the port it listens on. Currently the
+   * port can either be configured via the deprecated `port` configuration, or via the `listeners`
+   * configuration.
+   *
+   * This method uses `Application.parseListeners()` from `rest-utils` to get a list of listeners, and
+   * returns the port of the first listener to be used for the instance's identity.
+   *
+   * In theory, any port from any listener would be sufficient. Choosing the first, instead of say the last,
+   * is arbitrary.
+   */
+  // TODO: once RestConfig.PORT_CONFIG is deprecated, remove the port parameter.
+  static int getPortForIdentity(int port, List<String> configuredListeners) {
+    List<URI> listeners = Application.parseListeners(configuredListeners, port,
+            Arrays.asList(SchemaRegistryConfig.KAFKASTORE_SECURITY_PROTOCOL_SSL,
+                    SchemaRegistryConfig.KAFKASTORE_SECURITY_PROTOCOL_PLAINTEXT),
+            SchemaRegistryConfig.KAFKASTORE_SECURITY_PROTOCOL_PLAINTEXT);
+    return listeners.get(0).getPort();
```

Comment:
So, first, is the first listener necessarily the one you want? With http and https maybe it doesn't matter too much? Basically, I'm wondering if you want requests between the schema registry instances to use http (maybe you have them locked down/isolated via the network) but you listen on https listeners too, then the first listener isn't guaranteed to be the one you want to use, right? Or do we just need to add that to the config docs for listeners?

Second, I'm now wondering -- if you use only https, did we make sure forwarding of requests use https? I think in KafkaSchemaRegistry we just set `masterRestService` to use `http` with the master identity's host + port. We might need a system test to validate forwarding works in SSL-only multi-node SR clusters...